tests/Application/InstanceTypeExtractorTest.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
+
+use DataValues\DataValue;
+use DataValues\DecimalValue;
+use DataValues\MonolingualTextValue;
+use DataValues\QuantityValue;
+use DataValues\StringValue;
+use DataValues\TimeValue;
+use DataValues\UnboundedQuantityValue;
+use Elasticsearch\Endpoints\Cluster\State;
+use PHPUnit\Framework\TestCase;
+use Wikibase\DataModel\Entity\EntityIdValue;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\NumericPropertyId;
+use Wikibase\DataModel\Snak\PropertyNoValueSnak;
+use Wikibase\DataModel\Snak\PropertySomeValueSnak;
+use Wikibase\DataModel\Snak\PropertyValueSnak;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @covers \ProfessionalWiki\WikibaseFacetedSearch\Application\InstanceTypeExtractor
+ */
+class InstanceTypeExtractorTest extends TestCase {
+
+	private const INSTANCE_TYPE_ID = 'P42';
+
+	public function testReturnsInstanceTypeIdWithSingleInstanceTypeStatement(): void {
+		$statements = new StatementList(
+			$this->newStatement( 'P100', new StringValue( 'Foo' ) ),
+			$this->newInstanceTypeIdStatement( 'Q1' ),
+			$this->newStatement( 'P200', new StringValue( 'Bar' ) )
+		);
+
+		$this->assertHasInstanceType( 'Q1', $statements );
+	}
+
+	private function newStatement( string $propertyId, DataValue $value ): Statement {
+		return new Statement(
+			new PropertyValueSnak(
+				new NumericPropertyId( $propertyId ),
```

Comment:
Looks funny that `NumericPropertyId` gets a string

Reply:
The constructor takes a string.

Reply:
I know. No issue here. Just looks funny